From af0e950d3377b816f6cd19e9df0147ab331bc8e8 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Fri, 16 Aug 2019 21:16:57 +0200
Subject: [PATCH] Don't use sudo or virtualenv to install skills

---
 msm/skill_entry.py | 15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/msm/skill_entry.py b/msm/skill_entry.py
index 04a360f..293dfc6 100644
--- a/msm/skill_entry.py
+++ b/msm/skill_entry.py
@@ -270,28 +270,17 @@ class SkillEntry(object):
             constraints = DEFAULT_CONSTRAINTS
 
         LOG.info('Installing requirements.txt for ' + self.name)
-        can_pip = os.access(dirname(sys.executable), os.W_OK | os.X_OK)
-        pip_args = [sys.executable, '-m', 'pip', 'install']
+        pip_args = [sys.executable, '-m', 'pip', 'install', '--target=' + os.path.expanduser('~/.mycroft/opt/skills/deps')]
         pip_args += self.dependent_python_packages
         if constraints:
             pip_args += ['-c', constraints]
 
-        if not can_pip:
-            pip_args = ['sudo', '-n'] + pip_args
-
         with self.pip_lock:
             proc = Popen(pip_args, stdout=PIPE, stderr=PIPE)
             pip_code = proc.wait()
         if pip_code != 0:
             stderr = proc.stderr.read().decode()
-            if pip_code == 1 and 'sudo:' in stderr and pip_args[0] == 'sudo':
-                raise PipRequirementsException(
-                    2, '', 'Permission denied while installing pip '
-                           'dependencies. Please run in virtualenv or use sudo'
-                )
-            raise PipRequirementsException(
-                pip_code, proc.stdout.read().decode(), stderr
-            )
+            raise PipRequirementsException(pip_code, proc.stdout.read().decode(), stderr)
 
         return True
 
-- 
2.22.0

